contracts/core/src/contract.rs

```diff
@@ -1428,6 +1424,36 @@ pub fn migrate(
         }
         PAUSE.save(deps.storage, &new_pause_state)?;
 
+        #[cosmwasm_schema::cw_serde]
+        pub struct OldConfig {
+            pub factory_contract: Addr,
+            pub base_denom: String,
+            pub remote_denom: String,
+            pub idle_min_interval: u64,
+            pub unbonding_period: u64,
+            pub unbonding_safe_period: u64,
+            pub unbond_batch_switch_time: u64,
+            pub pump_ica_address: Option<String>,
+            pub transfer_channel_id: String,
+            pub emergency_address: Option<String>,
+            pub icq_update_delay: u64,
+        }
+
+        let old_config = Item::<OldConfig>::new("config").load(deps.storage)?;
+        let new_config = Config {
+            factory_contract: old_config.factory_contract,
+            remote_denom: old_config.remote_denom,
+            base_denom: old_config.base_denom,
+            emergency_address: old_config.emergency_address,
+            pump_ica_address: old_config.pump_ica_address,
+            idle_min_interval: old_config.idle_min_interval,
+            unbonding_period: old_config.unbonding_period,
+            icq_update_delay: old_config.icq_update_delay,
+            unbond_batch_switch_time: old_config.unbond_batch_switch_time,
+            unbonding_safe_period: old_config.unbonding_safe_period,
+        };
+        CONFIG.save(deps.storage, &new_config)?;
```

Comment:
Can we save CONFIG variable under new name like `config_v1`? To revert changes in case of emergency. We can remove old config during next migration (next contract version of this works as expected)